### PR TITLE
feat: compress url

### PR DESCRIPTION
**Pull request by alexander-akait**
**Description:**

Port - https://github.com/cssnano/cssnano/tree/master/packages/postcss-normalize-url

**BREAKING CHANGE:**

No

**Related issue (if exists):**

No

### CODE REVIEW

File: crates/swc_css_minifier/src/compress/url.rs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+            if !modifiers.is_empty() {
+                return;
+            }
+        }
+
+        if let Some(UrlValue::Str(Str { value, span, .. })) = &url.value {
+            let mut escaped = String::new();
+            let mut has_escaped = false;
+
+            for char in value.chars() {
```
Comment (kdy1) [inline crates/swc_css_minifier/src/compress/url.rs:28]: Please rename this to `c`, for better readability.

File: crates/swc_css_minifier/src/compress/url.rs [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+                    '(' | ')' | '"' | '\'' => {
+                        if has_escaped {
+                            return;
+                        }
+
+                        has_escaped = true;
+                        escaped.push('\\');
+                        escaped.push(char)
+                    }
+                    _ if char::is_whitespace(char) => {
```
Comment (kdy1) [inline crates/swc_css_minifier/src/compress/url.rs:39]: This can be `c.is_whitespace()`

File: crates/swc_css_minifier/src/compress/url.rs [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+                    _ => escaped.push(char),
+                }
+            }
+
+            url.value = Some(UrlValue::Raw(UrlValueRaw {
+                span: *span,
+                before: "".into(),
+                after: "".into(),
+                value: escaped.clone().into(),
+                raw: escaped.clone().into(),
```
Comment (kdy1) [inline crates/swc_css_minifier/src/compress/url.rs:57]: Is this clone required?

**Review by kdy1 — CHANGES_REQUESTED** [review]

**Comment by alexander-akait** [comment]
@kdy1 Rebased

**Review by kdy1 — DISMISSED** [review]
Thanks!


---


swc-bump:
 - swc_css_minifier

**Comment by kdy1** [comment]
Oh CI failed

**Comment by alexander-akait** [comment]
Give me sec, I will check

**Comment by alexander-akait** [comment]
Found small problem, WIP

**Comment by alexander-akait** [comment]
@kdy1 Fixed

**Comment by kdy1** [comment]
Thank you!

**Review by kdy1 — APPROVED** [review]

**Pull request merged**